### ISSUE TITLE
Dependencies, that have `react-native` as a dependency, resolve to one version of `react-native`

### ISSUE THREAD

**Issue by todorone** [mmazzarolo/react-native-universal-monorepo#22]
Thanks for such a great monorepo setup

It works great out of the box, but as soon as i started to add dependencies, that use `react-native`, it stopped working for me.

- mobile(depends on `react-native@0.65.1`)/reanimated/react-native => resolves to `react-native@0.65.1`
- tv(depends on `react-native-tvos@0.64.2`)/reanimated/react-native => resolves to `react-native@0.65.1` instead of `react-native-tvos@0.64.2`

I tried all combination of `nohoist` and `resolutions`, but i was not able to set version of nested `react-native` dependency of dependencies of platforms. Any hints?

**Comment by likern** [comment]
@todorone Looks like this is exactly problem I described above. I briefly looks at solving this problem. 
And I'm sure there is no solution (hope to be wrong). 

The only way to fix this is to use `postinstall` script which will delete all `react-native` packages under nested `node_modules` - this will require node to look `react-native` in above folders, thus it could find correct version.

**Comment by serge-sky** [comment]
here is nice and easy metro.config alternative to work with yarn v1.
https://gist.githubusercontent.com/Titozzz/66a5fce69cfbf93e488350551454aa7c/raw/b689adb35310fb5bb09d501fe18c4d235b6c091e/metro.config.js

Took from this article https://engineering.brigad.co/react-native-monorepos-code-sharing-f6c08172b417

**Comment by mmazzarolo** [comment]
@serge-sky , yeah, seems similar to the fix I mentioned above 👍 PRs to the monorepo tools are welcome!

**Comment by likern** [comment]
Share my experience with `yarn v3`. It works great, everything is smooth, even consuming native dependencies. Until today I was very happy.

**But** today I tried to patch native dependency which uses JSI C++ code https://github.com/ospfranco/react-native-quick-sqlite and **it's completely broken**.

**Native dependencies require exact package location** (since they require native parts in C++ using "../../react-native/..." paths).
Symlinking them makes packages broken (they can't find react-native using these paths since packages are symlinked).
**And Yarn v3 gives no control on how to fix this.**

https://github.com/ospfranco/react-native-quick-sqlite/blob/e736497166481bbb14a6a9da1df6691ccbee1204/android/CMakeLists.txt#L37

But the story is even worse 😢  **I can't even switch back to yarn classic** for the exact repository.
Even if I directly clone `yarn@1.22.10` by `npm pack` and run `node ./yarn.js -- --version` it gives incorrect (globally installed) version `3.0.2`. So **yarn v3 completely broke everything in that regard**. Of course, neither I can easily switch back to yarn classic completely 😠 (since using yarn v3 features).

I just can't now develop native code (the happy part it's not mandatory for me).
